### DISCUSSION TITLE
Better support for monorepos: allow sub configs

### DISCUSSION THREAD

**Discussion by pdonorio** [renovatebot/renovate#27153] [Suggest an Idea]
### Tell us more.

We love renovate but we have a big monorepo that mixes Business units, Guilds and languages.

The renovate config grew big and when there's a (new) problem it takes a lot of headaches and time to debug.

* Every new config or issue impacts everyone
* A single run is ~30min
* Logs are hard to read in general so after 30min run it's even harder to spot what has value

We didn't see any guide or question specifically in this direction so forgive if we didn't see the proper existing parameter/approach.

In case, one possible solution is to allow configurations per sub-folders (like you would have one per different repo). 

Also the github action that we use refers only to one configuration path.
Can it be using a matrix instead and we specify all the sub configs paths?

**Comment by rarkins** [comment]
Do you mean you run Renovate separately per group/folder and each of those Renovate runs essentially doesn't know the other exists?

**Reply by pdonorio** [reply]
Nope, we currently run a single config for all the repo.

> run Renovate separately per group/folder

can we do that?

**Reply by rarkins** [reply]
I was asking in the future tense. Can you describe in more detail how your ideal solution works?

**Reply by pdonorio** [reply]
Ok, nice :) 

In that case yes, we would run different configurations files for the a set of subfolders.

They don't need to really know about the other, but I imagine easily they could share some common configuration, although duplicating that would not be a problem.

**Reply by pdonorio** [reply]
Since we rely on the github action, it would be pretty handy to leverage the existing [matrix API](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy) to have one single workflow looking for multiple configuration files in different parallel runs.

**Reply by pdonorio** [reply]
Not sure how much can I expand more and still keep it generic, let me know if you have more questions 🙏

**Reply by viceice** [reply]
> Since we rely on the github action, it would be pretty handy to leverage the existing [matrix API](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy) to have one single workflow looking for multiple configuration files in different parallel runs.

that won't work

**Comment by not7cd** [comment]
If you are using multiple languages in a monorepo. What about running against a single config but with different managers enabled? If you are self-hosting ENV should do it.
https://docs.renovatebot.com/configuration-options/#enabledmanagers

**Reply by pdonorio** [reply]
We're running github workflow in our self hosted runners, so this can be possible.

Would we then be creating branches with the same github user?
Any issue that this might be creating for the (unified) dashboard?

**Reply by viceice** [reply]
It's not official supported to run multiple renovate instances with different config against a single repo.
This can cause a lot unwanted issues and we probably won't fix those.

**Reply by pdonorio** [reply]
Ok, that's the understanding so far, and it's also why we proposed here the idea.
